docs/CIP/CIP-XXXX/README.md

```diff
@@ -0,0 +1,272 @@
+---
+CIP: 1
+Title: Event-driven, wallet-opimised API
+Status: Proposed
+Category: Wallets
+Authors:
+  - William Wolff <william.wolff@iohk.io>
+  - Sefa Irken <sefa.irken@iohk.io>
+  - Rhys Bartels-Waller <rhys.bartelswaller@iohk.io>
+  - Martynas Kazlauskas <martynas.kaslauskas@iohk.io>
+  - Daniele Ricci <daniele.ricci@iohk.io>
+Implementors: N/A
+Solution-To: CPS-????
+Discussions:
+Created: 2024-07-10
+License: CC-BY-4.0
+---
+
+## Abstract
+
+We propose an event-driven, push-based API optimized for wallet applications, leveraging the principle that a wallet's state should be a pure function of the blockchain state. This approach eliminates polling strategies and aggregating views, enabling real-time synchronization with the blockchain as new blocks are appended. Our solution addresses key challenges in current wallet implementations, including rollback handling, address discovery and multi-account support.
+
+## Motivation: why is this CIP necessary?
+
+The CIP provides a partial solution to the problems described in [CPS-????](../CPS/CPS-XXXX/README.md). In particular, we strive to define a standardized approach serving data to edge clients in an unopinionated, close to blockchain native format using a push-based protocol.
+
+## Specification
+
+### Introduction
+
+#### What does a wallet need to construct transactions?
+
+- past transactions (receiving & spending) to derive balance and UTxO set
+- rewards (staking, voting)
+- network, era or epoch specific data, like:
+  - protocol parameters (tx fees, plutus cost models, ...)
+  - era summaries (slot length)
+  - genesis (security param)
+- the current tip/ block height for validity intervals of transactions as well as sync progress
+
+#### How do we serve this data?
+
+Through a push-based API, wallets are provided with a continuous stream of **relevant** on-chain events, enabling them to derive their current wallet state efficiently. This API supports various blockchain models, including UTXO-based chains and account-based chains. Additionally, clients receive secondary data relevant to the blockchain network they are subscribed to, ensuring comprehensive and up-to-date information.
+
+#### Protocol
+
+We use a bidirectional protocol (websockets) and encode all messages as JSON objects, while transactions contained in messages are preserved in the chain's native encoding standard (cbor). Messages adhere to a specific structure, ensuring consistency and facilitating efficient communication within our event-driven protocol. As we strive to support multiple blockchains, a server-sent message will always reference the originating chain it relates to.
```

Comment:
```suggestion
We use a bidirectional protocol and encode all messages as JSON objects, while transactions contained in messages are preserved in the chain's native encoding standard (cbor). Messages adhere to a specific structure, ensuring consistency and facilitating efficient communication within our event-driven protocol. As we strive to support multiple blockchains, a server-sent message will always reference the originating chain it relates to.
```

I feel it's better to not specify the exact protocol here and just describe it from a high level, so that we could see implementations in other protocols, such as gRPC.